interface/modules/custom_modules/oe-module-faxsms/src/Controller/RCFaxClient.php

```diff
@@ -0,0 +1,678 @@
+        }
+        try {
+            // do the request
+            $type = \GuzzleHttp\Psr7\MimeType::fromFilename(basename($file));
+            if (empty($type)) {
+                $type = $mime;
+            }
+            $request = $this->rcsdk->createMultipartBuilder()
+                ->setBody(array(
+                    'to' => array(
+                        array(
+                            'phoneNumber' => $phone,
+                            'name' => $name)
+                    ),
+                    'faxResolution' => 'High',
+                    'coverPageText' => text($comments)
+                ))
+                ->add($content, $file, array('Content-Type' => (string)$type))
+                ->request('/account/~/extension/~/fax');
+
+            $response = $this->platform->sendRequest($request);
+        } catch (ApiException $e) {
+            $message = $e->getMessage();
+            return text('Error: ' . $message) . PHP_EOL;
+        }
+
+        return xlt('Fax Successfully Sent');
+    }
+
+    /**
+     * @return bool|string
+     */
+    public function getUser(): bool|string
+    {
+        $id = $this->getRequest('uid');
+        $query = "SELECT * FROM users WHERE id = ?";
+        $result = sqlStatement($query, array($id));
+        $u = array();
+        foreach ($result as $row) {
+            $u[] = $row;
+        }
+        $u = $u[0];
+        $r = array($u['fname'], $u['lname'], $u['fax']);
+
+        return json_encode($r);
+    }
+
+    /**
+     * @return string
+     */
+    public function getNotificationLog(): string
+    {
+        $type = $this->getRequest('type');
+        $fromDate = $this->getRequest('datefrom');
+        $toDate = $this->getRequest('dateto');
+
+        try {
+            $query = "SELECT notification_log.* FROM notification_log WHERE notification_log.dSentDateTime > ? AND notification_log.dSentDateTime < ?";
+            $res = sqlStatement($query, array($fromDate, $toDate));
+            $row = array();
+            $cnt = 0;
+            while ($nrow = sqlFetchArray($res)) {
+                $row[] = $nrow;
+                $cnt++;
+            }
+            $responseMsgs = '';
+            foreach ($row as $value) {
+                $adate = ($value['pc_eventDate'] . '::' . $value['pc_startTime']);
+                $pinfo = str_replace("|||", " ", $value['patient_info']);
+                $msg = htmlspecialchars($value["message"], ENT_QUOTES);
+
+                $responseMsgs .= "<tr><td>" . $value["pc_eid"] . "</td><td>" . $value["dSentDateTime"] . "</td><td>" . $adate . "</td><td>" . $pinfo . "</td><td>" . $msg . "</td></tr>";
+            }
+        } catch (\Exception $e) {
+            $message = $e->getMessage();
+            return 'Error: ' . $message . PHP_EOL;
+        }
+
+        return $responseMsgs;
+    }
+
+    /**
+     * @return bool|string
+     */
+    public function getCallLogs(): bool|string
```

Comment:
Should this function get an acl check (also would consider for other applicable functions in these classes since looks like they are all available via the controller)